quickwit-doc-mapper/src/default_doc_mapper/default_mapper.rs

```diff
@@ -354,15 +377,18 @@ pub struct DefaultDocMapper {
     pub timestamp_field_name: Option<String>,
     /// Sort field name and order.
     pub sort_by: SortBy,
-    /// List of field mappings which defines how a json field is mapped to index fields.
-    pub field_mappings: FieldMappingEntry,
+    /// Root field mapping.
+    pub root_field_mapping: FieldMappingEntry,
     /// Schema generated by the store source and field mappings parameters.
     #[serde(skip_serializing)]
     schema: Schema,
     /// List of field names used for tagging.
     pub tag_field_names: BTreeSet<String>,
     /// Demux field name.
     pub demux_field_name: Option<String>,
+    /// List of fast fields. This list is used to reject Document if they
+    /// are missing a fast field.
+    pub required_fast_fields: Vec<FieldPath>,
```

Comment:
I don't understand this, why do we require fast fields to be set. fast fields can handle non-existent values

Reply:
Not in quickwit apparently.

Reply:
AFAIK, if there is a missing value, tantivy will put a default value in the fast field... 0 for integer/float. As I did not want a user to have unexpected results on aggregations, I rejected missing values.

Reply:
Generally speaking I prefer the spec in quickwit than the one in tantivy. I like the cardinality stuff too.

Reply:
Ok, I opened https://github.com/quickwit-oss/tantivy/issues/1338 on tantivy to track the problem of null values, I think this should be fixed on tantivy side.

Reply:
Thanks!

Reply:
> AFAIK, if there is a missing value, tantivy will put a default value in the fast field... 0 for integer/float. As I did not want a user to have unexpected results on aggregations, I rejected missing values.

That's a good point, I think we should improve tantivy here, so the user doesn't need to provide a value for null. We can e.g. compute it or handle it with a bitvec